flow360/component/flow360_params/solvers.py

```diff
@@ -547,6 +547,12 @@ class TurbulenceModelSolverLegacy(TurbulenceModelSolver, LegacyModel):
     rotation_correction: Optional[bool] = pd.Field(alias="rotationCorrection")
 
     def update_model(self):
+        if self.reconstruction_gradient_limiter is None:
+            if self.model_type == "SpalartAllmaras":
```

Comment:
Is there a model object type we can compare against? 

We should only define the mapping from model type string to object type in 1 place and then have the object type used everywhere